decide/src/main/java/org/example/Point.java

```diff
@@ -16,4 +16,45 @@ public double distance(Point other) {
     public static double triangleArea(Point p1, Point p2, Point p3) {
         return Math.abs(p1.x * (p2.y - p3.y) + p2.x * (p3.y - p1.y) + p3.x * (p1.y - p2.y)) / 2.0;
     }
+
+    /**
+     * Computes the angle from which Point p is the vertex.
+     * @param p2 first outer point of the angle
+     * @param p3 second outer point of the angle
+     * @return the angle formed by (p,p2) and (p,p3) in rad
+     */
+    public double angle(Point p2, Point p3) {
+        double scalarProduct = (p2.x - x) * (p3.x - x) + (p2.y - y) * (p3.y - y);
+        return Math.acos(scalarProduct / (distance(p2) * distance(p3)));
+    }
+
+    /**
+     * Computes the radius of the smallest circle containing all three points (p1, p2, p3)
+     * If one angle of the triangle is obtuse:
+     * the smallest circle containing the triangle has the opposite side as diameter
+     * Else:
+     * the smallest circle is the circumscribed circle.
+     * the radius is computed thanks to the law of sines : R = a * b * c / (4 * S)
```

Comment:
From this comment it looks like the formula belongs to the law of sines, but from what I can tell this is the formula for a circumscribed circle. Might be worth just clarifying this

Reply:
Yes, what I meant is that we use the law of sines to compute the radius of the circumscribed circle but you're right I'll change that